File: (8 kyu) Even or Odd/(8 kyu) Even or Odd.py
""" Even or Odd (8 kyu)
https://www.codewars.com/kata/even-or-odd
Create a function that takes an integer as an argument and returns
"Even" for even numbers or "Odd" for odd numbers.
"""


def even_or_odd(number):
    # #1
    # if number % 2 == 0:
    #     return "Even"
    # else:
    #     return "Odd"
    # #2
    # return "Odd" if number % 2 else "Even"
    # #3
    return ("Odd", "Even")[number % 2]

File: (8 kyu) Even or Odd/(8 kyu) Even or Odd.java
// #1
public class EvenOrOdd {
    public static String even_or_odd(int number) {
        if (number % 2 == 0) {
          return "Even";
        } else {
          return "Odd";
        }
    }
}

// #2
public class EvenOrOdd {
    public static String even_or_odd(int number) {
        return number % 2 == 0 ? "Even" : "Odd";
    }
}

// #3
public class EvenOrOdd {
    public static String even_or_odd(int number) {
        return (number & 1) == 0 ? "Even" : "Odd";
    }
}
